src/program.fs

```diff
@@ -7,48 +7,53 @@ Core abstractions for creating and running the dispatch loop.
 
 namespace Elmish
 
-open System
-
 /// Program type captures various aspects of program behavior
 type Program<'arg, 'model, 'msg, 'view> = {
     init : 'arg -> 'model * Cmd<'msg>
     update : 'msg -> 'model -> 'model * Cmd<'msg>
     subscribe : 'model -> Cmd<'msg>
-    view : 'model -> Dispatch<'msg> -> 'view
-    setState : 'model -> Dispatch<'msg> -> unit
+    view : Dispatch<'msg> -> 'model -> 'view
+    setState : Dispatch<'msg> -> 'model -> unit
     onError : (string*exn) -> unit
 }
 
 /// Program module - functions to manipulate program instances
 [<RequireQualifiedAccess>]
 [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
 module Program =
+    open Fable.Core
     open Fable.Core.JsInterop
     open Fable.Import.JS
 
     let internal onError (text: string, ex: exn) = console.error (text,ex)
 
     /// Typical program, new commands are produced by `init` and `update` along with the new state.
-    let mkProgram 
-        (init : 'arg -> 'model * Cmd<'msg>) 
+    let mkProgram
+        (init : 'arg -> 'model * Cmd<'msg>)
         (update : 'msg -> 'model -> 'model * Cmd<'msg>)
-        (view : 'model -> Dispatch<'msg> -> 'view) =
+        (view : Dispatch<'msg> -> 'model -> 'view) =
         { init = init
           update = update
           view = view
-          setState = fun model -> view model >> ignore
+          setState = fun dispatch ->
```

Comment:
This change isn't absolutely necessary but it show how the setState should be implemented by the various `withXXX`.